src/Sarif.UnitTests/TestData/SarifVersionOneToCurrentVisitor/ExpectedOutputs/OneRunWithRules.sarif

```diff
@@ -10,30 +10,32 @@
       "results": [
         {
           "ruleId": "C2001",
+          "ruleIndex": 0,
           "message": {
             "arguments": [ "someVariable" ],
             "messageId": "default"
           }
         },
         {
           "ruleId": "C2001",
+          "ruleIndex": 0,
           "message": {
             "arguments": [ "anotherVariable" ],
             "messageId": "default"
           }
         },
         {
-          "ruleId": "C2002-1",
+          "ruleId": "C2002",
```

Comment:
>C2002 [](start = 21, length = 5)

See the comment in the visitor code. It was never right to assign a `ruleKey` to a `ruleId`.

Reply:
In this case, in the input, the `ruleKey` did not point to any entry in the `rules` dictionary, so this `result` has no `ruleIndex` property (it defaults to -1).

---
In reply to: [250300944](https://github.com/Microsoft/sarif-sdk/pull/1232#discussion_r250300944) [](ancestors = 250300944)